index.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+/**
+ * Local
+ */
+
+const checkFile = require('./lib/check-file');
+const renderFile = require('./lib/render-file');
+const saveFile = require('./lib/save-file');
+
+/**
+ * Main export
+ */
+
+module.exports = function plugin(opts) {
+  opts = opts || {};
+
+  // Parse options and set defaults
+  const pattern = opts.pattern || '**';
+  const options = opts.options || {};
+
+  // Main plugin function
+  return function (files, metalsmith, done) {
+    // Get data from current metalsmith instance
+    const metadata = metalsmith.metadata();
+    const source = metalsmith.source();
+
+    // Loop through all files
+    Object.keys(files).forEach((filename) => {
+      // Check file, skip processing if it fails
+      if (!checkFile(filename, pattern)) return done();
+
+      // Transform file
+      const newname = renderFile(files, filename, metadata, source, options);
```

Comment:
It would be better if we aren't modifying files from the render function as it adds unnecessary complexion.
But I think we can change that in another pr.